flink-dist/src/main/flink-bin/bin/taskmanager.sh

```diff
@@ -63,7 +57,7 @@ if [[ $STARTSTOP == "start" ]]; then
         #
         TM_MAX_OFFHEAP_SIZE="8388607T"
 
-        if [[ "${STREAMINGMODE}" == "batch" ]] && useOffHeapMemory; then
+        if [[ "${FLINK_TM_MEM_ALLOC}" == "eager" ]] && useOffHeapMemory; then
```

Comment:
For the off-heap memory case, the JVM heap memory should be reduced regardless of eager or lazy memory allocation. I think this can be changed to

```bash
if useOffHeapMemory; then
```